test/cpp_extensions/rng_extension.cpp

```diff
@@ -47,7 +47,7 @@ Generator createTestCPUGenerator(uint64_t value) {
   return at::make_generator<TestCPUGenerator>(value);
 }
 
-Generator identity(Generator g) {
+Generator& identity(Generator& g) {
```

Comment:
What's going on here?

Reply:
In [`test/test_cpp_extensions_aot.py`](https://github.com/pytorch/pytorch/blob/233192be7334ab1dda217472f67d1de5bcb6685b/test/test_cpp_extensions_aot.py#L161-L163) there is the following code:

```py
copy2 = rng_extension.identity(copy)
self.assertEqual(rng_extension.getInstanceCount(), 1)
self.assertEqual(gen, copy2)
```

If we directly return a `Generator`, a separate Python object will be created. Although both of them refer to the same `GeneratorImpl`, they're not the same Python object and `assertEqual` will fail because we didn't implement comparison for `torch.Generator`. OTOH, accepting and returning a reference to the Generator will tell pybind11 to reuse the same Python object for `g`, thus returning the same object and passing the test.